great_expectations/checkpoint/actions.py

```diff
@@ -1078,15 +1110,18 @@ class SNSNotificationAction(ValidationAction):
     sns_message_subject: Optional[str]
 
     @override
-    def v1_run(self, checkpoint_result: CheckpointResult) -> str:
-        return send_sns_notification(
+    def v1_run(
+        self, checkpoint_result: CheckpointResult, action_context: ActionContext | None = None
+    ) -> dict:
+        msg = send_sns_notification(
             sns_topic_arn=self.sns_topic_arn,
             sns_subject=self.sns_message_subject or checkpoint_result.name,
             validation_results=json.dumps(
                 [result.to_json_dict() for result in checkpoint_result.run_results.values()],
                 indent=4,
             ),
         )
+        return {"result": msg}
```

Comment:
Let's just consistently return dicts (I can try and ensure a standardized typeddict or dataclass too)